digitalocean/vpcpeering/resource_vpcpeering.go

```diff
@@ -0,0 +1,212 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceDigitalOceanVPCPeering() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDigitalOceanVPCPeeringCreate,
+		ReadContext:   resourceDigitalOceanVPCPeeringRead,
+		UpdateContext: resourceDigitalOceanVPCPeeringUpdate,
+		DeleteContext: resourceDigitalOceanVPCPeeringDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the VPC Peering",
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The name of the VPC Peering",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"vpc_ids": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The list of VPCs to be peered",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Set: schema.HashString,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the VPC Peering",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time when the VPC was created",
+			},
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(2 * time.Minute),
+			Delete: schema.DefaultTimeout(2 * time.Minute),
+		},
+	}
+}
+
+func resourceDigitalOceanVPCPeeringCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(*config.CombinedConfig).GodoClient()
+
+	name := d.Get("name").(string)
+	vpcIDs := d.Get("vpc_ids").(*schema.Set).List()
+
+	vpcIDsString := make([]string, len(vpcIDs))
+	for i, v := range vpcIDs {
+		vpcIDsString[i] = v.(string)
+	}
+
+	vpcPeeringRequest := &godo.VPCPeeringCreateRequest{
+		Name:   name,
+		VPCIDs: vpcIDsString,
+	}
+
+	log.Printf("[DEBUG] VPC Peering create request: %#v", vpcPeeringRequest)
+
+	err := retry.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *retry.RetryError {
+		vpcPeering, resp, err := client.VPCs.CreateVPCPeering(context.Background(), vpcPeeringRequest)
+		if err != nil {
+			if resp != nil && resp.StatusCode == http.StatusForbidden {
+				return retry.RetryableError(err)
+			}
+			return retry.NonRetryableError(fmt.Errorf("error creating VPC Peering: %s", err))
+		}
+
+		d.SetId(vpcPeering.ID)
+
+		log.Printf("[DEBUG] Waiting for VPC Peering (%s) to become active", d.Get("name"))
+		stateConf := &retry.StateChangeConf{
+			Delay:      5 * time.Second,
+			Pending:    []string{"PROVISIONING"},
+			Target:     []string{"ACTIVE"},
+			Refresh:    vpcPeeringStateRefreshFunc(client, d.Id()),
+			Timeout:    10 * time.Minute,
```

Comment:
I think this timeout should be less than or equal to https://github.com/digitalocean/terraform-provider-digitalocean/pull/1160/files#diff-9f6d953dd67504f834d1166d111e56b71895731f5d87fe7fcfbcb6f94a5fe373R63 this one